code/modules/library/library_equipment.dm

```diff
@@ -0,0 +1,412 @@
+
+/obj/machinery/bookbinder/attack_ai(mob/user)
+	return attack_hand(user)
+
+/obj/machinery/bookbinder/attack_hand(mob/user)
+	if(..())
+		return
+	ui_interact(user)
+
+/obj/machinery/bookbinder/attack_ghost(mob/user)
+	ui_interact(user)
+
+/obj/machinery/bookbinder/attackby(obj/item/I, mob/user)
+	if(default_unfasten_wrench(user, I))
+		power_change()
+		return
+	if(istype(I, /obj/item/paper))
+		select_paper(I)
+	if(istype(I, /obj/item/paper_bundle))
+		select_paper_stack(I)
+	if(istype(I, /obj/item/book))
+		select_book(I)
+	else
+		return ..()
+
+/obj/machinery/bookbinder/proc/select_paper(obj/item/paper/P)
+	selected_content.title = P.name
+	selected_content.author = null
+	selected_content.content = list(P.info)
+
+/obj/machinery/bookbinder/proc/select_paper_stack(obj/item/paper_bundle/P)
+	selected_content.title = P.name
+	selected_content.author = null
+	selected_content.content = list()
+	for(var/obj/item/paper/I in P.contents)
+		selected_content.content += I.info
+
+/obj/machinery/bookbinder/proc/select_book(obj/item/book/B)
+	if(!B || B.protected || B.carved)
+		return
+	selected_content.title = B.title
+	selected_content.author = B.author
+	selected_content.author = B.summary
+	selected_content.content = B.pages
+	for(var/c in B.categories)
+		if(c)
+			selected_content.categories += c
+
+/obj/machinery/bookbinder/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "BookBinder", name, 700, 400, master_ui, state)
+		ui.open()
+
+/obj/machinery/bookbinder/ui_data(mob/user)
+	var/list/data = list()
+
+	var/list/selected_book_data = list(
+		"title" = selected_content.title ? selected_content.title : "not specified",
+		"author" = selected_content.author ? selected_content.author : "not specified",
+		"summary" = selected_content.summary ? selected_content.summary : "no summary",
+		"copyright" = selected_content.copyright ? selected_content.copyright : FALSE,
+		"categories" = selected_content.categories ? selected_content.categories : list()
+	)
+	data["selectedbook"] = selected_book_data
+	data["modal"] = ui_modal_data(src)
+	return data
+
+/obj/machinery/bookbinder/ui_static_data(mob/user)
+	var/list/static_data = list()
+
+	static_data["book_categories"] = list()
+	for(var/datum/library_category/category in GLOB.library_catalog.categories)
+		var/category_info = list(
+			"category_id" = category.category_id,
+			"description" = category.description,
+		)
+		static_data["book_categories"] += list(category_info)
+
+	return static_data
+
+/obj/machinery/bookbinder/ui_act(action, list/params, datum/tgui/ui)
+	if(..())
+		return
+
+	if(ui_act_modal(action, params))
+		return
+
+	switch(action)
+		if("print_book")
+			if(!printing)
+				printing = TRUE
+				visible_message("[src] begins to hum as it warms up its printing drums.")
```

Comment:
Missing spans

Reply:
fixed, I will do a final lookover over every file for these